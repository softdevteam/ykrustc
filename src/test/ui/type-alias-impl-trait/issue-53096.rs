#![feature(const_impl_trait, const_fn_fn_ptr_basics, rustc_attrs)]
// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(impl_trait_in_bindings, type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete
//[full_tait]~| WARN incomplete

type Foo = impl Fn() -> usize;
const fn bar() -> Foo { || 0usize }
const BAZR: Foo = bar();
//[min_tait]~^ ERROR not permitted here

#[rustc_error]
fn main() {} //[full_tait]~ ERROR
