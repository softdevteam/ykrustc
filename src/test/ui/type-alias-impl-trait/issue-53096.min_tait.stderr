error[E0658]: type alias impl trait is not permitted here
  --> $DIR/issue-53096.rs:10:19
   |
LL | const BAZR: Foo = bar();
   |                   ^^^^^
   |
   = note: see issue #63065 <https://github.com/rust-lang/rust/issues/63065> for more information
   = help: add `#![feature(impl_trait_in_bindings)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
